### PR TITLE
New Module: NextGenMap

### PR DESCRIPTION
**Pull request by cmatKhan**
[Nextgenmap](https://github.com/Cibiv/NextGenMap) is genomic aligner from [CIBIV](http://www.cibiv.at/). It is useful for identifying SNPs.

- [ x] This comment contains a description of changes (with reason).
- [ x] If you've fixed a bug or added code that should be tested, add tests!
- [ x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ x] If necessary, include test data in your PR.
- [ x] Remove all TODO statements.
- [ x] Emit the `versions.yml` file.
- [ x] Follow the naming conventions.
- [ x] Follow the parameters requirements.
- [ x] Follow the input/output options guidelines.
- [ x] Add a resource `label`
- [ x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nextgenmap/main.nf [hunk 1]
```diff
@@ -0,0 +1,58 @@
+process NEXTGENMAP {
+    tag "$meta.id"
+    label 'process_medium'
```
Comment (SPPearce) [inline modules/nextgenmap/main.nf:3]: I would expect process_high for an aligner, as they generally scale linearly with number of cpus.

**Review by SPPearce — APPROVED** [review]
Everything else looks good to me.

File: modules/nextgenmap/main.nf (hunk 1 above)
Comment (cmatKhan) [reply modules/nextgenmap/main.nf:3]: I wish there was a `high_cpu_low_mem` label. The amount of memory that `process_high` requests is overkill for this program.

`npm` is quite fast. The time issue that I run into most on my cluster with this program is not the time it takes to align, but the time it takes to wait in the queue for resources. But, I defer to other opinions on this. Admittedly, I work mostly on smaller genomes.

Comment (SPPearce) [reply modules/nextgenmap/main.nf:3]: I see that bbmap is also on process_medium, so there is precedent.

Comment (SPPearce) [reply modules/nextgenmap/main.nf:3]: I'm going to leave it at process_medium and get it merged.

**Pull request merged**